pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix

```diff
@@ -318,7 +318,8 @@ in rec {
         mv $NIX_BUILD_TOP/temp "$PWD/deps/${pname}"
         cd $PWD
 
-        ln -s ${deps}/deps/${pname}/node_modules "deps/${pname}/node_modules"
+        cp -r "${deps}/deps/${pname}/node_modules" "deps/${pname}/node_modules"
+        chmod -R +w "deps/${pname}/node_modules"
```

Comment:
Do you know which dependency tries to write to the node_modules ?
It might be good to add it as a comment, just to know in case they get rid of the dependency then we can get rid of that line.

Reply:
Yeah that's an unfortunate hack. After some investigation I believe it is this bug: https://github.com/babel/babel/issues/6906
It got fixed in [Babel 7.7.0 onwards](https://github.com/babel/babel/commit/487f10f84d03a2019c7aa8da254577478bb28e3b) but to me it looks like we still use older versions in Mobilizon.

I think best we can do currently is add a comment and revert the change eventually.

```
mobilizon>  ERROR  Failed to compile with 8 errors          5:54:19 PM                                                                                                                                                              
mobilizon>  error  in ../../node_modules/@absinthe/socket-apollo-link/node_modules/@babel/runtime/helpers/newArrowCheck.js                                                                                                          
mobilizon> Syntax Error: Error: EACCES: permission denied, mkdir '/tmp/nix-build-mobilizon-2.0.2.drv-0/mobilizon-js-src/deps/mobilizon/node_modules/.cache'                                                                         
mobilizon>  error  in ../../node_modules/@absinthe/socket/node_modules/@babel/runtime/helpers/newArrowCheck.js                                                                                                                      
mobilizon> Syntax Error: Thread Loader (Worker 5)                                                                                                                                                                                   
mobilizon> EACCES: permission denied, mkdir '/tmp/nix-build-mobilizon-2.0.2.drv-0/mobilizon-js-src/deps/mobilizon/node_modules/.cache'
mobilizon>  error  in ../../node_modules/@absinthe/socket/node_modules/@babel/runtime/helpers/objectSpread.js
mobilizon> Syntax Error: Thread Loader (Worker 3)
mobilizon> EACCES: permission denied, mkdir '/tmp/nix-build-mobilizon-2.0.2.drv-0/mobilizon-js-src/deps/mobilizon/node_modules/.cache'
mobilizon>  error  in ../../node_modules/@absinthe/socket/node_modules/@babel/runtime/helpers/objectWithoutProperties.js
mobilizon> Syntax Error: Thread Loader (Worker 4)
mobilizon> EACCES: permission denied, mkdir '/tmp/nix-build-mobilizon-2.0.2.drv-0/mobilizon-js-src/deps/mobilizon/node_modules/.cache'
mobilizon>  error  in ../../node_modules/@babel/runtime/helpers/esm/classCallCheck.js
mobilizon> Syntax Error: Thread Loader (Worker 2)
mobilizon> EACCES: permission denied, mkdir '/tmp/nix-build-mobilizon-2.0.2.drv-0/mobilizon-js-src/deps/mobilizon/node_modules/.cache'
mobilizon>  error  in ../../node_modules/@babel/runtime/helpers/esm/createClass.js
mobilizon> Syntax Error: Thread Loader (Worker 3)
mobilizon> EACCES: permission denied, mkdir '/tmp/nix-build-mobilizon-2.0.2.drv-0/mobilizon-js-src/deps/mobilizon/node_modules/.cache'
mobilizon>  error  in ../../node_modules/@babel/runtime/helpers/esm/defineProperty.js
mobilizon> Syntax Error: Thread Loader (Worker 1)
mobilizon> EACCES: permission denied, mkdir '/tmp/nix-build-mobilizon-2.0.2.drv-0/mobilizon-js-src/deps/mobilizon/node_modules/.cache'
mobilizon>  error  in ../../node_modules/@babel/runtime/helpers/esm/typeof.js
mobilizon> Syntax Error: Thread Loader (Worker 0)
mobilizon> EACCES: permission denied, mkdir '/tmp/nix-build-mobilizon-2.0.2.drv-0/mobilizon-js-src/deps/mobilizon/node_modules/.cache'
mobilizon>  ERROR  Error: Build failed with errors.
mobilizon> Error: Build failed with errors.
mobilizon>     at /nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/@vue/cli-service/lib/commands/build/index.js:207:23
mobilizon>     at /nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/webpack/lib/webpack.js:148:8
mobilizon>     at /nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/webpack/lib/HookWebpackError.js:68:3
mobilizon>     at Hook.eval [as callAsync] (eval at create (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
mobilizon>     at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/tapable/lib/Hook.js:18:14)
mobilizon>     at Cache.shutdown (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/webpack/lib/Cache.js:150:23)
mobilizon>     at /nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/webpack/lib/Compiler.js:1158:15
mobilizon>     at Hook.eval [as callAsync] (eval at create (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
mobilizon>     at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/tapable/lib/Hook.js:18:14)
mobilizon>     at Compiler.close (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/webpack/lib/Compiler.js:1151:23)
mobilizon>     at /nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/webpack/lib/webpack.js:147:16
mobilizon>     at finalCallback (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/webpack/lib/Compiler.js:436:32)
mobilizon>     at /nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/webpack/lib/Compiler.js:453:13
mobilizon>     at Hook.eval [as callAsync] (eval at create (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:33:1)
mobilizon>     at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/tapable/lib/Hook.js:18:14)
mobilizon>     at onCompiled (/nix/store/mpi7h7xjza4p4xx7m4vrxqrqgiprhx6p-mobilizon-modules-2.0.2/node_modules/webpack/lib/Compiler.js:451:21)
mobilizon> error Command failed with exit code 1.
```

Reply:
This is still a major blocker, in that it doesn't make sense to merge this PR as is. It breaks other stuff using yarn2nix.
I tried to track down how to get rid of `babel<7.7.0` and it looks like there are only two packages using it:

- https://github.com/absinthe-graphql/absinthe-socket
- https://github.com/jumpn/utils-graphql

Both packages haven't seen changes in years and I'm a bit scared of bumping their babel version as I somehow expect a lot of stuff to break.

Reply:
I plan to give `pluralsh-absinthe-socket-apollo-link` - which is [given in this issue](https://github.com/absinthe-graphql/absinthe-socket/pull/58) - a try, which may partly help here.

Reply:
@tcitworld That would be of huge help!